src/createRoom.ts

```diff
@@ -256,6 +256,17 @@ export default async function createRoom(opts: IOpts): Promise<string | null> {
     let modal;
     if (opts.spinner) modal = Modal.createDialog(Spinner, null, 'mx_Dialog_spinner');
 
+    if (opts.encryption && opts.createOpts?.invite?.length > 0) {
```

Comment:
I am not sure this is doing what you want.
Here the opts.encryption is already set, and downloading key wont change anything I think.
This need to be done earlier to decide or not to set the opts.encryption like done using `determineCreateRoomEncryptionOption(..)` or `canEncryptToAllUsers()`  before calling createRoom

Reply:
According to the issue it already creates an encrypted room. This decision is taken [here](https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/utils/dm/createDmLocalRoom.ts#L56). Following the code it already triggers key downloads somewhere. Only the first message is unencrypted sometimes. But it must be caused by something else then…

Reply:
The decision to encrypt messaging in room is based on the presence of a m.room.encryption state event in that room (`room.currentState.getStateEvents(EventType.RoomEncryption, "")`).
It might be a race when the state event is not yet synced backed by the crypto layer (`onCryptoEvent`) and you start typing.
It's with the new DM flow right?